service/gcs/bridge/bridge.go

```diff
@@ -317,7 +340,55 @@ func (b *Bridge) PublishNotification(n *prot.ContainerNotification) {
 	b.responseChan <- resp
 }
 
+// negotiateProtocol was introduced in v4 so will not be called
+// with a minimum lower than that.
+func (b *Bridge) negotiateProtocol(w ResponseWriter, r *Request) {
+	if r.Version != prot.PvInvalid {
```

Comment:
I think this is handled by the routing mechanism, right? Since `negotiateProtocol` is registered with `prot.PvInvalid`, it should only be called when that's set.

Reply:
Actually the goal here is to make sure that we don't ever try to negotiate a new version. IE: if the request has updated the version once this call would error out the 2nd time. And for a v3 workflow where a createContainer came in it would error to call it again.

Reply:
You win. Removed